pkg/block/mdfetcher.go

```diff
@@ -0,0 +1,103 @@
+package block
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+type MetadataFetcher struct {
+	cachedBlockMDs map[ulid.ULID]*metadata.Meta
+	rwLock         *sync.RWMutex
```

Comment:
Usually we just name it `mu`. `rw` is part of type so no need to put type info into variable. (: 

```suggestion
	mu         *sync.RWMutex
```